### DISCUSSION TITLE
SideSetsAroundSubdomainGenerator

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#23849] [Q&A General] [answered]
Dear MOOSE Team,
I used “SideSetsAroundSubdomainGenerator” (with block = block # of Block 2) to generate the boundary lines around Block 2 (please see the attached sketch).
It seems that only the Block 2 side of boundary lines are generated.
Namely, I cannot set “NeumannBC” on the Block 1 side of boundary lines.
I would like to apply “NeumannBC” on the Block 1 side of boundary lines as BC and Block 2 is isolated from Block 1.
Please help me generate the boundary lines on the Block 1 side.

Sincerely,
S. Thomas
[Boundary_Generation_Around_A_Block.pdf](https://github.com/idaholab/moose/files/11073714/Boundary_Generation_Around_A_Block.pdf)

**Comment by GiudGiud** [comment]
Hello

There's a few other options here. One that comes to mind, is "between block 1 and 2"
https://mooseframework.inl.gov/moose/source/meshgenerators/SideSetsBetweenSubdomainsGenerator.html

Guillaume

**Reply by styyokuda** [reply]
Thank you Guillaume.
The challenge here is to set different values of NeumannBC for top, bottom, left, and right sides of the boundary.
I believe "SideSetsBetweenSubdomainsGenerator" takes the same value of NeumannBC for all sides of boundary.
Is my understanding right?
Please help.

S. Thomas

**Reply by GiudGiud** [reply]
oh right
Can you try this one then? Are the normals all constant on the sides you care about?
https://mooseframework.inl.gov/moose/source/meshgenerators/SideSetsFromNormalsGenerator.html

**Reply by styyokuda** [reply]
Thank you very much Guillaume.  I will try.
Actually the values of NeumannBC for top, bottom, left, and right sides of the boundary are assigned after the calculations as the temperature dependence, I will modify NeumannBC.C and h for that. Yes, they are normal to the boundary lines from the Block 1 side.

Thank you very much again and I would like to let you know,.
S. Thomas

**Reply by GiudGiud** [reply]
What do you mean by after the calculation?
Do you mean that the boundary conditions on one domain depend on the variables in the other domain?

**Reply by styyokuda** [reply]
Yes, the value of flux is calculated from the concentration at the boundary and temperature.
I hope my idea works.

S. Thomas

**Reply by GiudGiud** [reply]
If flux, concentration and temperature
- are solved within the same input file
- influence each other on both sides of the interface (high C causes high flux for example, and high flux is a loss term in the C equation)
you may want to consider an interface kernel instead of a boundary condition.

**Reply by styyokuda** [reply]
I thought the assign flux boundary value only  on the Block 1 side and completely isolate the Block 2 side from the computation is more stable.  I don't know ...

Thank you very much for your time and consideration.
S. Thomas

**Reply by styyokuda** [reply]
If an interface kernel is used, does “SideSetsAroundSubdomainGenerator” work?
Please let me know.

S. Thomas

**Reply by GiudGiud** [reply]
No issue with sidesets for interface kernels. You just choose the variable and neibghbor variable according to the sideset orientation

**Reply by styyokuda** [reply]
Dear Guillaume,
"SideSetsAroundSubdomainGenerator" with "interface kernels" works.

I can set the interface direction from Block 2 to Block 1 by "SideSetsBetweenSubdomainsGenerator" but 
I cannot set the top, right, bottom, and left interfaces by "SideSetsBetweenSubdomainsGenerator".
I would still like to ask if there is a way to set flux on the top, right, bottom, and left interfaces on the Block 1 side without "interface kernels" because there is not enough mesh in Block 2 for the computation.
Please help.

S. Thomas

**Reply by GiudGiud** [reply, marked as answer]
Hello

The SideSetsBetweenSubdomainsGenerator has no way to select a normal
The SideSetsFromNormalsGenerator has no way to select a block
Your best chance will be
- https://mooseframework.inl.gov/source/meshgenerators/SideSetsFromBoundingBoxGenerator.html
and just draw tight boxes around the 4 sides you care about. There s parameters to select the direction of the sidesets

I think you've identified a need here, to either:
- be able to flip an existing sideset (and keep or not the original one)
- be able to restrict a sideset to a block you care about or a normal you care about

If you're open to developing the contribution of the capability would be welcome. If not I may have someone to help me on meshing this summer. 

Guillaume
